### PR TITLE
feat: add pixi support for dev environments

### PR DESCRIPTION
**Pull request by napetrov**
## Summary

Adds [pixi](https://pixi.sh) as a supported dev-environment manager, alongside the existing `pip install -e ".[dev]"` / conda-forge workflows described in `CONTRIBUTING.md`.

## Motivation

`CONTRIBUTING.md`'s "conda-forge" option already pointed contributors at conda-forge for `castxml`; pixi builds directly on the same channel and lets a single `pyproject.toml`-driven `pixi install && pixi run test` provision the Python dev tools *and* the system tools (`castxml`, a C/C++ compiler, `cmake`, `libabigail`, `abi-compliance-checker`) that the `integration`/`libabigail`/`abicc` marker lanes need — replacing several manual `conda create`/`apt install` steps with one command.

## Changes

- `pyproject.toml`: new `[tool.pixi.*]` tables —
  - `default` env: editable `abicheck` install + `dev`/`mcp` extras (lint, mypy, pytest). Tasks: `test`, `test-cov`, `lint`, `fmt`, `fmt-check`, `typecheck`, `check` (composite).
  - `integration` env (linux-64/osx-64/osx-arm64): adds `castxml` + a C/C++ compiler + `cmake`. Task: `test-integration`.
  - `parity` env (linux-64 only, matching conda-forge's platform support for these packages): adds `libabigail` + `abi-compliance-checker`. Tasks: `test-libabigail`, `test-abicc`.
  - `docs` env: `mkdocs` + plugins. Tasks: `docs-build` (`--strict`), `docs-serve`.
- `pixi.lock`: committed lockfile (resolved cross-platform: linux-64/osx-64/osx-arm64/win-64).
- `.gitignore`: ignore `.pixi/` (the local env/cache dir; `pixi.lock` itself is tracked).
- `CONTRIBUTING.md`: documents pixi as the new recommended setup option, with a task/environment reference table and a note that conda-forge tool versions can drift from what CI pins via apt/brew/choco.
- `CLAUDE.md`: one-line pointer to pixi as an alternative to `pip install -e ".[dev]"`.
- `CHANGELOG.md`: `[Unreleased]` entry.

No changes to the published package, its runtime dependencies, or any CI workflow — this is dev-tooling only.

## Verification

Installed `pixi` 0.73.0 locally and exercised all four environments end-to-end before pushing:
- `default`: `pixi run lint` / `typecheck` / `test` → 15588 passed, 14 skipped, 4 xfailed.
- `integration`: `castxml`/`gcc`/`g++`/`cmake` resolved from conda-forge; `test-integration` ran (some struct-layout/calling-convention edge cases differ from CI's pinned castxml/gcc versions — documented as a known conda-forge version-drift caveat, not a regression).
- `parity`: `abidiff`/`abi-compliance-checker` resolved; `test-abicc` fully green (24 passed, 1 skipped), `test-libabigail` mostly green (11/14 — same version-drift caveat, libabigail 2.9.0 from conda-forge vs. whatever `apt-get install abigail-tools` pins in CI).
- `docs`: `mkdocs build --strict` succeeded.

Also confirmed `ruff check`, `mypy abicheck/`, and `scripts/check_ai_readiness.py` show no findings beyond what's already present on `main` (verified via `git stash`).

## Checklist

- [x] Tests added / updated for new behaviour — N/A, dev-tooling only; existing suite re-run under pixi as verification (see above).
- [x] `CHANGELOG.md` updated (under `[Unreleased]`)
- [x] Docs updated if user-visible behaviour changed — `CONTRIBUTING.md`/`CLAUDE.md` updated; no user-visible (published-package) behaviour changed.
- [x] No new `mypy` or `ruff` errors introduced

---
_Generated by [Claude Code](https://claude.ai/code/session_013vpbQuPuHu3VXK6Tbm5QHT)_

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -228,3 +228,89 @@ paths_to_mutate = [
...
+[tool.pixi.feature.parity.dependencies]
+castxml = "*"
+c-compiler = "*"
+cxx-compiler = "*"
+libabigail = "*"
+abi-compliance-checker = "*"
+
+[tool.pixi.feature.parity.tasks]
+test-libabigail = "pytest tests/test_abidiff_parity.py tests/test_abicompat_parity.py tests/test_abipkgdiff_parity.py -v -m libabigail --tb=short"
+test-abicc = "pytest tests/test_abicc_parity.py tests/test_xml_parity.py -v -m abicc --tb=short"
```
[the remark this answers is not in the document]
Comment (napetrov) [reply pyproject.toml]: Good catch — fixed in 5b2f96d: `test-libabigail`/`test-abicc` now use marker-scoped selection (`pytest tests/ -m libabigail` / `-m abicc`) instead of a hard-coded file list, so `test_abidiff_parity_extended.py`, `test_surface_scope_parity.py`, and `test_abicc_parity_extended.py` are picked up too.

---
_Generated by [Claude Code](https://claude.ai/code)_
